test/algorithms/area/area.cpp

```diff
@@ -175,6 +177,34 @@ void test_large_integers()
     BOOST_CHECK_CLOSE(int_area, double_area, 0.0001);
 }
 
+struct precise_cartesian : bg::strategies::detail::cartesian_base
+{
+    template <typename Geometry>
+    static auto area(Geometry const&)
+    {
+        return bg::strategy::area::precise_cartesian<>();
+    }
+};
+
+void test_inaccurate_cases()
+{
+    typedef bg::model::point<double, 2, bg::cs::cartesian> point_type;
+
+    bg::model::polygon<point_type> poly;
+
+    bg::read_wkt("POLYGON((0.10000000000000001 0.10000000000000001,\
+                           0.20000000000000001 0.20000000000000004,\
+                           0.79999999999999993 0.80000000000000004,\
+                           1.267650600228229e30 1.2676506002282291e30,\
+                           0.10000000000000001 0.10000000000000001))", poly);
+
+    double inaccurate = bg::area(poly);
+    double accurate = bg::area(poly, precise_cartesian());
+
+    BOOST_CHECK_CLOSE(inaccurate, 0, 0.0001);
+    BOOST_CHECK_CLOSE(accurate, -0.315, 0.0001);
```

Comment:
Quite a difference. Is the polygon indeed anti clockwise?

Reply:
It depends on how you perform the computation. With double arithmetic the area algorithm computes 4 determinants with the following values
```
-0.030000000000000012768
-0.5999999999999999778
-1606938044258989383512154151116236941093472087848332596084736
1606938044258989383512154151116236941093472087848332596084736
```
which then has to sum up and divide by 2. The implementation before this PR computes `0` while after this PR  `-0.315` which is clearly more accurate. 

Now computing the determinants accurately is a different story, not related to this PR. Note that doing this computation with boost multi-precision the sign of the area is positive so the polygon is CW oriented. I added another example where both precise area an multi-precision returns positive sign while the old area strategy returns `0` to emphasize the usefulness of performing a sum more accurately.

Reply:
I see. Good, thanks.